gradle/dependencies.gradle

```diff
@@ -15,367 +15,386 @@
+libraries.jcip =                'net.jcip:jcip-annotations'
+libraries.jgit =                'org.eclipse.jgit:org.eclipse.jgit'
+libraries.joda =                'joda-time:joda-time'
+libraries.jsch =                'com.jcraft:jsch'
+libraries.jsr305 =              'com.google.code.findbugs:jsr305'
+libraries.junit =               'junit:junit'
+libraries.kryo =                'com.esotericsoftware.kryo:kryo'
+libraries.maven3 =              'org.apache.maven:maven-core'
+libraries.maven3_wagon_file =   'org.apache.maven.wagon:wagon-file'
+libraries.maven3_wagon_http =   'org.apache.maven.wagon:wagon-http'
+libraries.nativePlatform =      'net.rubygrapefruit:native-platform'
+libraries.nekohtml =            'net.sourceforge.nekohtml:nekohtml'
+libraries.objenesis =           'org.objenesis:objenesis'
+libraries.plexus_container =    'org.codehaus.plexus:plexus-container-default'
+libraries.plist =               'com.googlecode.plist:dd-plist' // for XCode IDE integration support
+libraries.pmaven_common =       'org.sonatype.pmaven:pmaven-common'
+libraries.pmaven_groovy =       'org.sonatype.pmaven:pmaven-groovy'
+libraries.rhino =               'org.mozilla:rhino'
+libraries.simple =              'org.simpleframework:simple'
+libraries.testng =              'org.testng:testng'
+libraries.xerces =              'xerces:xercesImpl'
+libraries.xmlApis =             'xml-apis:xml-apis'
+libraries.slf4j_api =           'org.slf4j:slf4j-api'
+libraries.jcl_to_slf4j =        'org.slf4j:jcl-over-slf4j'
+libraries.jul_to_slf4j =        'org.slf4j:jul-to-slf4j'
+libraries.log4j_to_slf4j =      'org.slf4j:log4j-over-slf4j'
 
-libraries.ant = 'org.apache.ant:ant:1.9.9'
-libraries.asm =  'org.ow2.asm:asm:6.0'
-libraries.asm_commons =  'org.ow2.asm:asm-commons:6.0'
-libraries.asm_util =  'org.ow2.asm:asm-util:6.0'
-libraries.asm_tree =  'org.ow2.asm:asm-tree:6.0'
-libraries.cglib = 'cglib:cglib:3.2.5'
-libraries.bndlib = 'biz.aQute.bnd:biz.aQute.bndlib:3.4.0'
-libraries.commons_cli = 'commons-cli:commons-cli:1.2'
-libraries.commons_io = "commons-io:commons-io:${versions.commons_io}"
-libraries.commons_lang = 'commons-lang:commons-lang:2.6'
-libraries.commons_collections = 'commons-collections:commons-collections:3.2.2'
-libraries.commons_compress = 'org.apache.commons:commons-compress:1.14'
-libraries.jsch = 'com.jcraft:jsch:0.1.54'
-libraries.fastutil = "it.unimi.dsi:fastutil:7.2.1"
+// the 'if' condition here is a configuration time performance optimization, that applied the constraints only to root projects of the subproject dependencies tree
+subprojects { project -> if (project.name in ['baseServices', 'cli']) {
+    configurations.whenObjectAdded {
+        if (name == 'compile') {
+            dependencies {
+                constraints {
+                    // lock dependencies that are part of the Gradle distribution
+                    compile(libraries.aether_connector) { version { strictly versions.aether } }
```

Comment:
Here's an example of a 'platform', where we want to ensure that all parts of `aether` have the same version. It would be good to think about how we can declare it in that way. (Nebula have the identical problem they are trying to solve)